### PR TITLE
Users create

### LINKED ISSUE #3: Create Users Endpoint
As a user, when I send a POST request to `api/v1/users` along with a body containing:
+ Name
+ Email address
+ Password
+ Password confirmation
then I receive a response containing a unique api_key.

*Request Format:*
```
POST /api/v1/users
Body:
    { 
      name: "Louis Armstrong",
      email: "trumpet@songs.jazz",
      password: "password",
      password_confirmation: "password",
     }
```
Response Format:
```
Status: 201
    { 
          api_key: "8675309b13bd98e"
     }
```

Note: A User will also have an api_key, api_key_active and will have many gardens

- [x] For Password use bcrypt gem
- [x] Properly formatted request returns 201
- [x] Custom Error Handling for edge cases

### PR DESCRIPTION
**Pull request by bplantico**
## Changes proposed in this pull request:
This PR adds the api/v1/users endpoint which receives a POST request along with a body and returns an api_key if the user is created. It also adds the User model which has many Gardens (a Garden belongs to a User) along with the accompanying migration.

## Card(s) closed in this pull request:
close #3 

### Model Test Coverage: 100%
- [x] Tested my new feature(s) as well as any feasible edge cases (if possible)
- [x] Checked coverage/index.html - did not add any new code that's not covered by testing (if possible)
- [x] Ran the test suite - all tests are passing
- [x] Merged in the latest master to my branch with git pull origin master & resolved merge conflicts
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class User < ApplicationRecord
+  before_create :set_api_key
```
Comment (Mycobee) [inline app/models/user.rb]: I like this way of setting an API key with the before action

File: app/controllers/api/v1/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      render json: UserSerializer.new(user), status: :ok
+    else
+      render json: "{ Please check that passwords match. }".to_json, status: :bad_request
+    end
+  end
+
+  private
+
+  def passwords_match
+    user_params[:password] == user_params[:password_confirmation]
```
[the remark this answers is not in the document]
Comment (bplantico) [reply app/controllers/api/v1/users_controller.rb]: @Mycobee Bcrypt authenticates a password vs the password_digest stored for a user, but it doesn't check if the password and password_confirmation match.
